devices/ePaper/Drivers/Gdew0154x/Gdew0154x.cs

```diff
@@ -0,0 +1,636 @@
+﻿// Copyright (c) 2022 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Drawing;
+using System.Threading;
+using Iot.Device.EPaper.Buffers;
+using Iot.Device.EPaper.Enums;
+using nanoFramework.UI;
+
+namespace Iot.Device.EPaper.Drivers.GDEW0154x
+{
+    /// <summary>
+    /// Base class for GDEW0154-based ePaper devices.
+    /// </summary>
+    public abstract class Gdew0154x : IEPaperDisplay
+    {
+        private SpiDevice _spiDevice;
+        private GpioController _gpioController;
+        private GpioPin _resetPin;
+        private GpioPin _busyPin;
+        private GpioPin _dataCommandPin;
+
+        private bool _disposed;
+
+        /// <summary>
+        /// The max supported clock frequency for the GDEW0154x controller. 10MHz.
+        /// </summary>
+        public const int SpiClockFrequency = 10_000_000;
+
+        /// <summary>
+        /// The supported <see cref="System.Device.Spi.SpiMode"/> by the GDEW0154x controller.
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Gdew0154x"/> class.
+        /// </summary>
+        /// <param name="spiDevice">The communication channel to the SSD1681-based dispay.</param>
+        /// <param name="resetPin">The reset GPIO pin. Passing an invalid pin number such as -1 will prevent this driver from opening the pin. Caller should handle hardware resets.</param>
+        /// <param name="busyPin">The busy GPIO pin.</param>
+        /// <param name="dataCommandPin">The data/command GPIO pin.</param>
+        /// <param name="width">The width of the display.</param>
+        /// <param name="height">The height of the display.</param>
+        /// <param name="gpioController">The <see cref="GpioController"/> to use when initializing the pins.</param>
+        /// <param name="enableFramePaging">Page the frame buffer and all operations to use less memory.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="spiDevice"/> is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Display width and height can't be less than 0 or greater than 200.</exception>
+        protected Gdew0154x(
+            SpiDevice spiDevice,
+            int resetPin,
+            int busyPin,
+            int dataCommandPin,
+            int width,
+            int height,
+            GpioController gpioController,
+            bool enableFramePaging = false)
+        {
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+            _gpioController = gpioController ?? throw new ArgumentNullException(nameof(gpioController));
+
```

Comment:
can you add a "_shouldDispose" to check if the gpio controller should be disposed? See the pattern here for example: https://github.com/nanoframework/nanoFramework.IoT.Device/blob/9ce639dc2f38b94ffe1f2cfb6606a180b97bfb23/devices/A4988/A4988%20.cs#L20
In short, you can pass a null GpioController and you'll create it, also you'll dispose it in this case. If it's not null and the shouldDispose is true, you'll dispose it otherwise no.